### PR TITLE
Fix Watcher HTTP connection config for longevity

### LINKED ISSUE #52997: Watcher webhooks fail with invalidated SSL sessions
**Elasticsearch version** : 7.5.2

**Description of the problem including expected versus actual behavior**:

Webhook actions to call remote services which are configured to fire infrequently (say every 30 mins) will alternate between success and failure. This was reported by a user when watcher was hosted in a cloud environment. A successful call is followed up by a failure which logs the fact that the [session is invalidated](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/java.base/share/classes/sun/security/ssl/SSLSessionImpl.java#L753) . This fails the user's action which is obviously less than great. Presumably this failure is also the cue for the Watcher client to drop the broken connection meaning the next triggered action will be forced to recreate the connection to the remote service. This would explain the flip-flop behaviour between successes and failures when triggering these actions.

**Steps to reproduce**:

A reproduction is possible using a self-managed Elasticsearch, running ES 7.5.2 on a macbook and configured with the following alert:

	{
	  "trigger": {
		"schedule": {
		  "interval": "30m"
		}
	  },
	  "input": {
		"simple": {}
	  },
	  "condition": {
		"always": {}
	  },
	  "actions": {
		"telegram": {
		  "webhook": {
			"scheme": "https",
			"host": "api.telegram.org",
			"port": 443,
			"method": "post",
			"path": "/MYUUID/sendMessage",
			"params": {
			  "text": "hello world",
			  "chat_id": "-1"
			},
			"headers": {}
		  }
		}
	  }
	}

Next step, wait for the watcher to fire. And once it has fired, suspend laptop. After a while, unsuspend the laptop, and wait for the watcher to fire, and the first execution will fail with a reset connection.

**Resolution**
Ideally an invalidated session would be something a webhook action should attempt to recover from automatically with failing the user action.

### PR DESCRIPTION
**Pull request by DaveCTurner**
Watcher uses a connection pool for outgoing HTTP traffic, which means
that some HTTP connections may live for a long time, possibly in an idle
state. Such connections may be silently torn down by a remote device, so
that when we re-use them we encounter a `Connection reset` or similar
error.

This commit introduces a setting allowing users to set a finite expiry
time on these connections, and also enables TCP keepalives on them by
default so that a remote teardown will be actively detected sooner.

Closes #52997

### CODE REVIEW

File: x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/common/http/HttpClientTests.java [hunk 1]
```diff
@@ -770,6 +770,50 @@ public void testCreateUri() throws Exception {
         assertCreateUri("https://example.org", "");
     }
 
+    public void testConnectionReuse() throws Exception {
```
Comment (DaveCTurner) [inline x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/common/http/HttpClientTests.java:773]: This was the least flaky way I could think of to test this but it's not exactly watertight, we cannot 100% guarantee that the connection will be re-used in the first case, nor can we guarantee that the client port won't be re-used in the second case.

I also couldn't work out a way to test that TCP keepalives are enabled at all.

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/2

**Comment by DaveCTurner** [comment]
@elasticmachine update branch

**Review by jakelandis — APPROVED** [review]
Thanks David, these changes look good and is much better then current workaround of [sending a connection closed header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection) to force non-persistent connections.

**Pull request merged**